lookout/style/format/quality_report.py

```diff
@@ -0,0 +1,104 @@
+import argparse
+from collections import Counter
+import glob
+import os
+import logging
+import sys
+
+import bblfsh
+from bblfsh.client import NonUTF8ContentException
+import numpy
+from sklearn.metrics import classification_report,confusion_matrix
+from tqdm import tqdm
+
+from lookout.core.api.service_data_pb2 import File
+from lookout.core.cmdline import ArgumentDefaultsHelpFormatterNoNone
+from lookout.core import slogging
+from lookout.style.format.features import FeatureExtractor, CLASSES
+from lookout.style.format.model import FormatModel
+
+
+def prepare_files(folder, client, language):
+    files = []
+
+    if not folder.endswith("**/*"):
+        folder = os.path.join(folder, "**/*")
+
+    # collect filenames with full path
+    filenames = glob.glob(folder, recursive=True)
+
+    for file in tqdm(filenames):
+        if not os.path.isfile(file):
+            continue
+        try:
+            res = client.parse(file, language)
+        except NonUTF8ContentException:
+            # skip files that can't be parsed because of UTF-8.
+            continue
+        if res.status == 0 and res.language.lower() == language.lower():
+            uast = res.uast
+            path = file
+            with open(file) as f:
+                content = f.read().encode("utf-8")
+            files.append(File(content=content, uast=uast, path=path))
+    return files
+
+
+def quality_report(args):
+    _log = logging.getLogger("QualityReport")
+    _log.setLevel(args.log_level)
+
+    client = bblfsh.BblfshClient(args.bblfsh)
+    files = prepare_files(args.input, client, args.language)
+    _log.info("Number of files: %s" % (len(files)))
+
+    fe = FeatureExtractor(language=args.language)
+    X, y, nodes = fe.extract_features(files)
+
+    analyzer = FormatModel().load(args.model)
+    rules = analyzer._rules_by_lang[args.language]
+    y_pred = rules.predict(X)
+
+    target_names = [CLASSES[cls_ind] for cls_ind in numpy.unique(y)]
+    _log.info("\n" + classification_report(y, y_pred, target_names=target_names))
```

Comment:
same as for visualization, should probably be printed directly since it's the main goal of the program

Reply:
Yep